mobile_verifier/src/unique_connections/ingestor.rs

```diff
@@ -0,0 +1,181 @@
+{
+    pub async fn create_managed_task(
+        pool: PgPool,
+        settings: &Settings,
+        file_upload: FileUpload,
+        file_store: FileStore,
+        authorization_verifier: AV,
+    ) -> anyhow::Result<impl ManagedTask> {
+        let (verified_unique_connections, verified_unique_conections_server) =
+            VerifiedUniqueConnectionsIngestReportV1::file_sink(
+                settings.store_base_path(),
+                file_upload.clone(),
+                FileSinkCommitStrategy::Manual,
+                FileSinkRollTime::Default,
+                env!("CARGO_PKG_NAME"),
+            )
+            .await?;
+
+        let (unique_connections_ingest, unique_connections_server) =
+            file_source::Continuous::msg_source::<UniqueConnectionsIngestReport, _>()
+                .state(pool.clone())
+                .store(file_store.clone())
+                .lookback(LookbackBehavior::StartAfter(settings.start_after))
+                .prefix(FileType::UniqueConnectionsReport.to_string())
+                .create()
+                .await?;
+
+        let radio_threshold_ingestor = Self::new(
+            pool.clone(),
+            unique_connections_ingest,
+            verified_unique_connections,
+            authorization_verifier,
+        );
+
+        Ok(TaskManager::builder()
+            .add_task(verified_unique_conections_server)
+            .add_task(radio_threshold_ingestor)
+            .add_task(unique_connections_server)
+            .build())
+    }
+
+    pub fn new(
+        pool: PgPool,
+        unique_connections_receiver: Receiver<FileInfoStream<UniqueConnectionsIngestReport>>,
+        verified_unique_connections_sink: FileSinkClient<VerifiedUniqueConnectionsIngestReportV1>,
+        authorization_verifier: AV,
+    ) -> Self {
+        Self {
+            pool,
+            unique_connections_receiver,
+            verified_unique_connections_sink,
+            authorization_verifier,
+        }
+    }
+
+    async fn run(mut self, shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tracing::info!("starting unique connections ingestor");
+        loop {
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => break,
+                Some(file) = self.unique_connections_receiver.recv() => {
+                    self.process_unique_connections_file(file).await?;
+                }
+            }
+        }
+        tracing::info!("stopping unique connections ingestor");
+        Ok(())
+    }
+
+    async fn process_unique_connections_file(
+        &self,
+        file_info_stream: FileInfoStream<UniqueConnectionsIngestReport>,
+    ) -> anyhow::Result<()> {
+        let mut txn = self.pool.begin().await?;
+
+        let mut stream = file_info_stream.into_stream(&mut txn).await?;
+        while let Some(unique_connections_report) = stream.next().await {
+            let verified_report_status = self
```

Comment:
Noting it here, whether it should apply or not is up for debate...the known carrier key check can now be performed at the ingestor side and if invalid/unknown the UniqueConnectionsReport would be dropped to the floor at that point.  If that is used the verifier would simply act as a consumer of the ingest reports without a need to validate again and without a need to write out Verified versions of the ingest report.

Unless the verifier has to apply its own verifications ( other than the known carrier key check ) I am not convinced that the pattern implemented here and elsewhere provides any real additional value

Reply:
at the moment it is consistent will all other data that verifier reads.  The only data that is verified by the ingestor is never read by the verifier.  I think consistency here is important

Reply:
the consistency argument is fair